scripts/root-ca-retriever/public-certs.ts

```diff
@@ -0,0 +1,414 @@
+/**
+ * Copyright 2025 Defense Unicorns
+ * SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+ */
+
+import * as https from "https";
+import * as fs from "fs";
+import * as path from "path";
+import * as yaml from "js-yaml";
+import * as crypto from "crypto";
+import { parse as parseCSV } from "csv-parse/sync";
+
+export interface PublicCACert {
+  commonName: string;
+  content: string;
+  certificateIssuerOrganization?: string;
+  geographicFocus?: string;
+  owner?: string;
+  companyWebsite?: string;
+}
+
+interface PublicCATrustConfig {
+  include: PublicCACert[];
+  exclude: PublicCACert[];
+}
+
+interface MozillaCSVRecord {
+  "Common Name or Certificate Name": string;
+  "Certificate Issuer Organization": string;
+  "Certificate Issuer Organizational Unit": string;
+  "Geographic Focus": string;
+  Owner: string;
+  "Company Website": string;
+  [key: string]: string; // Allow for other fields
+}
+
+const PUBLIC_CA_CERTS_URL = "https://curl.se/ca/cacert.pem";
+const MOZILLA_CSV_URL =
+  "https://ccadb.my.salesforce-sites.com/mozilla/IncludedRootCertificateReportCSVFormat";
+const TARGET_PUBLIC_CERT_DIR = "public"; // subdirectory in certs directory to put public CA certs
+
+/**
+ * Retrieves public CA certificates bundle from curl.se (Mozilla's curated list)
+ * and returns the raw PEM content as a string
+ * @returns Promise that resolves to the raw PEM bundle content
+ * @throws {Error} When HTTP request fails or returns non-200 status code
+ */
+export async function retrievePublicCACertificates(): Promise<string> {
+  console.log("Starting public CA certificate download...");
+
+  return new Promise<string>((resolve, reject) => {
+    console.log(`Downloading from: ${PUBLIC_CA_CERTS_URL}`);
+
+    https
+      .get(PUBLIC_CA_CERTS_URL, response => {
+        if (response.statusCode !== 200) {
+          reject(
+            new Error(`Failed to get public CA certificates. Status code: ${response.statusCode}`),
+          );
+          return;
+        }
+
+        let data = "";
+        response.on("data", chunk => {
+          data += chunk;
+        });
+
+        response.on("end", () => {
+          console.log("Downloaded public CA certificates");
+          resolve(data);
+        });
+      })
+      .on("error", err => {
+        reject(err);
+      });
+  });
+}
+
+/**
+ * Downloads and parses Mozilla's Common CA Database (CCDB) CSV data containing
```

Comment:
Should this be `CCADB`?
```suggestion
 * Downloads and parses Mozilla's Common CA Database (CCADB) CSV data containing
```

Reply:
fixed